tools/federation/src/main/java/org/eclipse/rdf4j/federated/repository/FedXRepositoryConfig.java

```diff
@@ -152,6 +163,11 @@ public Resource export(Model m) {
 			m.add(implNode, DATA_CONFIG, vf.createLiteral(getDataConfig()));
 		}
 
+		if (getConfig() != null) {
```

Comment:
This will mean: any externally set `FedXConfig` (e.g. via code) will be serialized into the repository configuration, e.g. when saving it.

I think overall (from history) there is a bit of technical design issue in the `FedXRepositoryConfig` and `FedXConfig`: the `FedXConfig` mixes "string" like options and things that can only be set via code (e.g. the TaskWrapper). 

For our product use-case: we are having an extension of FedXConfig (with some proprietary options, and also a TaskWrapper) and we are setting this config via code. We generally would not want to see it exported to the repository config.

So maybe to retain compatibility on that level, we can use one of the following options:

a) introduce another boolean field (say `isExternalConfig`)  and only export if it is not externally provided
b) put the handling of the FedXConfig for both parsing and exporting to a protected method, which can be overridden in customizations of the repository config

I think option b) is cleaner an will allow most flexibility and still retain compatibility.

So the idea would be to move your new exporting/parsing part of the code from this class to new protected methods

* `protected void exportFedXConfig(Model model, Resource implNode)`
* `protected void parseFedXConfig(Model model, Resource implNode)`

and call those in the current places.

Does this make sense to you?

Reply:
Yes - that sounds good. I'll have a go.

Reply:
On second thought - the proposed method signatures allow sub-classes to do whatever they want (be it config-related or not). Would it not make more sense to pass them the node which contains the config (like is done in the separate class) - separation of concerns and all that.
And e.g. allow `new FedXConfig(ConfigReader)` to inject a custom or sub-classed implementation?

(But I also appreciate that this probably doesn't need to be over-engineered.)

Reply:
5a160dfd1bfcaa48e3c5